sklearn/utils/tests/test_validation.py

```diff
@@ -539,3 +540,29 @@ def test_suppress_validation():
     assert_all_finite(X)
     sklearn.set_config(assume_finite=False)
     assert_raises(ValueError, assert_all_finite, X)
+
+
+class DummyMemory(object):
+    def __init__(self):
+        pass
+
+    def cache(self, func):
+        return func
+
+    cachedir = None
```

Comment:
I'm not sure what you mean. An example?

What I mean is that we might as well accept (and therefore test) objects that have a `cache` method which operates as expected, regardless of whether it's got all the other attributes that a `joblib.Memory` has.